src/Microsoft.ML.StandardLearners/FactorizationMachine/FactorizationMachineInterface.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Internal.CpuMath;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System.Runtime.InteropServices;
+
+using System.Security;
+
+namespace Microsoft.ML.Runtime.FactorizationMachine
+{
+    internal unsafe static class FieldAwareFactorizationMachineInterface
+    {
+        internal const string NativePath = "FactorizationMachineNative";
+        public const int CbAlign = 16;
+
+        private static bool Compat(AlignedArray a)
```

Comment:
Hi @sfilipi and @wschin , could I ask, was the usage of `AlignedArray` here intentional, in the sense that without aligned array something bad happens? Or was it just copying over the practices in some existing code? If the latter, could we just as easily have used `fixed` with simple arrays, and whatnot? Relevant to discussions I've had with @eerhardt ... we want to get rid of `AlignedArray`.

Reply:
In a few small benchmark performance tests I've run, `AlignedArray` performs worse than using a regular array.

Reply:
FFM's core computation is done by SSE code, which requires the memory blocks to be aligned. The main computation doesn't call any member functions of `AlighedArray`, we just use `AlighedArray` as something like `new` operator. Thus, the performance is possibly as good as regular array.

Reply:
Thanks for the note Eric. Added notes to Issue #146, to address this. 

---
In reply to: [198231400](https://github.com/dotnet/machinelearning/pull/383#discussion_r198231400) [](ancestors = 198231400)